public_html/wp-content/mu-plugins/blocks/assets/src/blocks-store.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * Implements wc-blocks-store.
+ */
+
+/**
+ * WordPress dependencies
+ */
+const { registerStore, dispatch } = wp.data;
+const { apiFetch } = wp;
+const { addQueryArgs } = wp.url;
+
+/**
+ * Define store name.
+ */
+export const WC_BLOCKS_STORE = 'wc-blocks-store';
+
+/**
+ * Initial state.
+ */
+const DEFAULT_STATE = {
+	loadingEntities : []
+};
+
+const MAX_POSTS = 100;
+
+/**
+ * Supported entities with their args.
+ */
+const API_ARGS = {
+	speakers: {
+		path: 'wp/v2/speakers',
+		query: {
+			orderby  : 'id',
+			order    : 'desc',
+			per_page : MAX_POSTS,
+			_embed   : true,
+			context  : 'view',
+		}
+	},
+	speakerTracks: {
+		path: 'wp/v2/session_track',
+		query: {
+			per_page: MAX_POSTS,
+			context: 'view',
+		}
+	},
+	speakerGroups: {
+		path: '/wp/v2/speaker_group',
+		query: {
+			orderby  : 'name',
+			order    : 'asc',
+			per_page : MAX_POSTS,
+		}
+	},
```

Comment:
`[MIGHT]` - I wonder if these parts should be registered by the corresponding blocks, so that the store isn't tightly coupled to things outside of it?

Reply:
So I was thinking blocks should be isolated from actually fetching data from API. They should work as if data is already available to them to isolate responsibilities properly, to that end it would make sense that queries are defined in store instead of individual blocks.

Another way to look at it is that, many blocks could need same data, so if we define API queries in blocks, either we will duplicate the queries, or will call a common method from those. Defining in store seems more cleaner, and we won't have to handle promise etc in the blocks. Also adding support of retries/pagination etc should be easy.

Reply:
Ah, that makes sense 👍